bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -141,9 +152,25 @@ public Builder setStatsLogger(StatsLogger statsLogger) {
             return this;
         }
 
+
+        public Builder dnsResolver(DNSToSwitchMapping dnsResolver) {
+            this.dnsResolver = dnsResolver;
+            return this;
+        }
+
+        public Builder requestTimer(HashedWheelTimer requestTimer) {
```

Comment:
@eolivelli I think if the client is embedded in other service, it might be allowed to share same hashwheeltimer with other component running in same jvm. in that sense, it might make sense to add this.